role/handle.go

```diff
@@ -132,3 +136,49 @@ func getResponseList(roles []model.DBRole) []model.Role {
 
 	return res
 }
+
+// CacheRoles ...
+func CacheRoles() {
+	var (
+		ctx = context.Background()
+		wg  sync.WaitGroup
+	)
+
+	// Find
+	roles := findByCondition(ctx, bson.M{})
```

Comment:
không nên làm ntn, vì call db khá nhiều

1. khai báo struct để lưu

```go
type CachedRoles struct {
  Role string `json:"role"`
  IsAdmin bool `json:"isAdmin"`
  Permissions []string `json:"permissions"`
}
```

2. dùng 2 query để lấy hết roles và permissions ra (dùng code sẽ tiết kiệm hơn query db nhiều lần)
3. for roles, for tiếp permissions để gom lại thành 1 mảng theo struct trên
4. lưu xuống mem cache

Reply:
done